src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -106,19 +106,23 @@ protected override void OnPersist(ApplicationEngine engine)
         {
             base.OnPersist(engine);
 
+            // Set next validators
+
+            StorageItem storage = engine.Snapshot.Storages.GetAndChange(CreateStorageKey(Prefix_NextValidators), () => new StorageItem());
+            storage.Value = GetValidators(engine.Snapshot).ToByteArray();
+        }
+
+        protected override void PostPersist(ApplicationEngine engine)
+        {
+            base.PostPersist(engine);
+
             // Distribute GAS for committee
 
             int index = (int)(engine.Snapshot.PersistingBlock.Index % (uint)ProtocolSettings.Default.CommitteeMembersCount);
             var gasPerBlock = GetGasPerBlock(engine.Snapshot);
-            var committee = GetCommitteeMembers(engine.Snapshot).ToArray();
-            var pubkey = committee.OrderBy(p => p).ElementAt(index);
+            var pubkey = GetCommittee(engine.Snapshot)[index];
             var account = Contract.CreateSignatureRedeemScript(pubkey).ToScriptHash();
             GAS.Mint(engine, account, gasPerBlock * CommitteeRewardRatio / 100);
```

Comment:
We set `nextValidators` before process the transactions and reward the committee after process these transactions, isn't it?

Reply:
Yes.

Reply:
Maybe we should wait for https://github.com/neo-project/neo/pull/1907 and reward every epoch (28 blocks)?

Reply:
Why #1907 is needed?

Reply:
Not needed, but ig #1907 it's merged, we can reward them every 28 blocks.